lib/vagrant-openshift/helper/command_helper.rb

```diff
@@ -160,7 +160,9 @@ def repo_checkout_bash_command(repo, url)
         git fetch origin
         git fetch --prune origin +refs/tags/*:refs/tags/*
         git reset --merge
-        git checkout #{branch}
+        if ! git checkout #{branch}; then
+          git checkout -b #{branch} origin/#{branch}
```

Comment:
I see, in that case we have both `upstream/pre-flight-checks` and `origin/pre-flight-checks`, and the same remote twice (with two different fetch protocols, too!). Do we want to change that logic or just layer your fix on top?